### DISCUSSION TITLE
Static/compiled version needed (w/o experience in compiling)

### DISCUSSION THREAD

**Discussion by RsQuser** [ImageMagick/ImageMagick#2174] [Help]
I'm new to the board, so please excuse me if I'm somehow unprecise. I would be lucky to find help here concerning the following problem:

I run an image database tool (based on PHP/MySQL) and have been using ImageMagick for a few years to create thumbnails, place watermarks, ...  Unfortunately I don't have any compiling skills AND use a shared host w/o root access. Therefore I used a somehow older version of IM (via "cmd("/usr/bin/convert ...") that was compiled once. It was running for years without problems.

A few days ago the host provider (Hosteurope) made OS updates - now IM does not work any longer. I canot figure out what exactly the reason is, but I somehow got the impression to need a new compiled/static IM to run in the updated environment.

Is there anyone I can contact for support/help concerning this problem (paid, of course)? Please feel free to contact me here or via mail (support@pictogrande.com). 

Best regards,

Stefan

(If I get it right, the forum https://www.imagemagick.org/discourse-server/ is no longer active? There was a section "consulting" for searching developer support.)

**Comment by fmw42** [comment]
Contact your hosting service and ask where ImageMagick is now located. Presumably the OS is Linux and it usually comes with ImageMagick. But it could have been installed in a different path than before. Typical paths are /usr/bin or /usr/local/bin.  But it is possible they have it installed, but have not put its path into the $PATH environment variable. You could run the following command in a PHP script and potentially find out its location.


```
<?php
echo "<pre>";
system("type -a convert");  
echo "</pre>";
?> 

```
or

```
<?php
echo "<pre>";
system("which -a convert");  
echo "</pre>";
?> 
```

But the best way is to contact your hosting company and ask them where it is located and why you cannot access it or how to access it.

**Comment by RsQuser** [comment]
Unfortunately, Hosteurope doesn't have IM installed by default. They had that a few years ago, but then cut it. ;-(

That's why I need to use a static version from"/cgi-bin". Changing to a provider that has IM isn't an option at the moment.

**Comment by fmw42** [comment]
Sorry, I cannot help any further. That is about all I know. One of the IM developers will have to answer your questions about installing the static version.  What is your platform/OS? Is it Linux or Windows? Can you not get your hosting service to install ImageMagick for you in your directory?

**Comment by RsQuser** [comment]
OS is "Linux wp501 4.14.132-xenu-he"

Hosteurope does not install Imagemagick at all.  ;-(( That's why I need a /cgi-bin/ standalone solution ...

**Comment by urban-warrior** [comment]
Try the portable version of ImageMagick @ https://imagemagick.org/download/binaries/magick.

**Comment by RsQuser** [comment]
Unfortunately this doesn't work, too. Errors that occur:

[0] => Cannot open /proc/self/exe: Permission denied
[1] => Failed to get fs offset for /proc/self/exe
